pkg/virt-launcher/virtwrap/access-credentials/access_credentials.go

```diff
@@ -320,12 +315,26 @@ func (l *AccessCredentialManager) agentSetAuthorizedKeys(domName string, user st
 		// Zero flags argument means that the authorized_keys file is overwritten with the authorizedKeys
 		return domain.AuthorizedSSHKeysSet(user, authorizedKeys, 0)
 	}()
-	if errors.Is(err, libvirt.ERR_NO_SUPPORT) {
-		// If AuthorizedSSHKeysSet method is not supported, use the old method
-		desiredAuthorizedKeys := strings.Join(authorizedKeys, "\n")
-		return l.agentWriteAuthorizedKeysFile(domName, user, desiredAuthorizedKeys)
+	if firstErr == nil {
+		return nil
 	}
-	return err
+
+	log.Log.V(4).Infof("Could not set SSH key using virDomainAuthorizedSSHKeysSet(): %v", firstErr)
+
+	// If AuthorizedSSHKeysSet method failed, use the old method
+	desiredAuthorizedKeys := strings.Join(authorizedKeys, "\n")
+	secondErr := l.agentWriteAuthorizedKeysFile(domName, user, desiredAuthorizedKeys)
+	if secondErr == nil {
+		return nil
+	}
+
+	log.Log.V(4).Infof("Could not set SSH key using guest-file-write: %v", secondErr)
```

Comment:
I am not sure this is needed as you return the error here.

Reply:
done